core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -715,17 +718,21 @@ class HttpGoogleServicesDAO(
   }
 
   def diagnosticBucketRead(userInfo: UserInfo, bucketName: String): Future[Option[ErrorReport]] = {
-    implicit val service = GoogleInstrumentedService.Storage
-    Future {
-      val getter = getStorage(getUserCredential(userInfo)).buckets().get(bucketName)
-      try {
-        blocking {
-          executeGoogleRequest(getter)
-        }
-        None
-      } catch {
-        case t: HttpResponseException => Some(ErrorReport(StatusCode.int2StatusCode(t.getStatusCode), t.getMessage))
+    // we make requests to the target bucket as the default pet, if the user only has read access.
+    // the default pet is created in a project without APIs enabled. Due to Google issue #16062674, we cannot
+    // call the GCS JSON API as the default pet; we must use the XML API. In turn, this means we cannot use
+    // the GCS client library (which internally calls the JSON API); we must hand-code a call to the XML API.
+
+    val bucketUrl = s"https://storage.googleapis.com/$bucketName"
+    val bucketRequest = httpClientUtils.addHeader(RequestBuilding.Head(bucketUrl), Authorization(userInfo.accessToken))
+
+    httpClientUtils.executeRequest(http, bucketRequest) map { httpResponse =>
+      httpResponse.status match {
+        case StatusCodes.OK => None
+        case x => Some(ErrorReport(x, x.defaultMessage()))
       }
+    } recover {
+      case t:Throwable => Some(ErrorReport(t))
```

Comment:
I intentionally used `HttpClientUtils` directly here instead of having `HttpGoogleServicesDAO` extend `DsdeHttpDAO`, to discourage other developers from making http calls instead of using Google client libs. This does mean there's some duplication between here and `DsdeHttpDAO`; happy to hear opinions this was the wrong choice.